python/graphstorm/sagemaker/utils.py

```diff
@@ -206,16 +233,40 @@ def download_graph(graph_data_s3, graph_name, part_id, local_path, sagemaker_ses
     """
     # Download partitioned graph data.
     # Each training instance only download 1 partition.
-    graph_config = f"{graph_name}.json"
     graph_part = f"part{part_id}"
 
     graph_path = os.path.join(local_path, graph_name)
     graph_part_path = os.path.join(graph_path, graph_part)
     os.makedirs(graph_path, exist_ok=True)
     os.makedirs(graph_part_path, exist_ok=True)
-    try:
-        S3Downloader.download(os.path.join(graph_data_s3, graph_config),
+
+    graph_data_s3 = graph_data_s3[:-1] if graph_data_s3.endswith('/') else graph_data_s3
+
+    # We split on '/' to get the bucket, as it's always the third split element in an S3 URI
+    s3_input_bucket = graph_data_s3.split("/")[2]
+    # Similarly, by having maxsplit=3 we get the S3 key value as the fourth element
+    s3_input_key = graph_data_s3.split("/", maxsplit=3)[3]
+
+    s3_client = boto3.client('s3')
+    for graph_config  in [f"{graph_name}.json", "metadata.json"]:
```

Comment:
Why f"{graph_name}.json" and "metadata.json" here?

Reply:
data_dispatch.py creates a file named `metadata.json`. I guess the single-instance partitioning instead creates `{graph_name}.json` so we should support both.